packages/node_modules/@ciscospark/internal-plugin-reaction/src/reaction.js

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {SparkPlugin} from '@ciscospark/spark-core';
+import uuid from 'uuid';
+
+const Reaction = SparkPlugin.extend({
+  namespace: 'Reaction',
+
+  /**
+  * helper function to create the request object
+  * @param {Object} options
+  * @returns {Object} the reaction request object
+  */
+  createReactionRequest(options) {
+    return {
+      method: 'POST',
+      service: 'conversation',
+      resource: '/activities',
+      options: options.options,
+      body:
+      {
+        clientTempId: uuid.v4(),
+        objectType: 'activity',
+        verb: options.verb,
+        actor: {
+          objectType: 'person',
+          id: options.participant.id,
+          entryUUID: options.participant.entryUUID
+        },
+        object: {
+          objectType: 'messageProperty',
+          tags: ['REACTIONS'],
+          id: options.activityID,
+          reactionType: options.reactionID
+        },
+        target: {
+          id: options.convoID,
+          url: options.targetURL,
+          objectType: 'conversation'
+        }
+      }
+    };
+  },
+
+  /**
+  * helper function to verify options parameter
+  * @param {Object} options
+  * @returns {string} the rejection error message or null if options is good
+  */
+  rejectReactionOptions(options) {
```

Comment:
I would expect this to be something like `validateReaction`, the name is a bit confusing to me.